File: Arrays & Hashing/0349. Intersection of Two Arrays/0349.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        unordered_set<int> visit;
        for (auto num : nums1) {
            visit.insert(num);
        }
        for (auto num : nums2) {
            if (visit.find(num) != visit.end()) {
                res.push_back(num);
                visit.erase(num);
            }
        } return res;
    }
};

File: Arrays & Hashing/0349. Intersection of Two Arrays/0349.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = []
        visit = set()
        for num in nums1:
            visit.add(num)
        for num in nums2:
            if num in visit:
                res.append(num)
                visit.remove(num)
        return res